### PR TITLE
Invite Participants

### PR DESCRIPTION
**Pull request by ihor-palii**
By default in RapidPro, users cannot send individual flow in a simple way to their contacts unless they are entering the flow and starting it for each user. To facilitate this process, we developed Invite Participants.

Details on:
https://app.asana.com/0/1159876841343611/1158809372498665/f

### CODE REVIEW

**Comment by teehamaral** [comment]
@ihor-palii @stoneric - Per Jacob, we have a client using this feature, so I reopened this PR.

File: templates/contacts/contact_invite_participants.haml [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+                          -if field.show_in_table
+                            %td.field
+                              {{ object|contact_field:field.key }}
+
+                        %td.value-received.field_received
+                          %nobr
+                            {% short_datetime object.created_on %}
+                        
+                        %td.invite-column.value-received.field_received
+                          %a.invite.btn.btn-primary.btn-block{href:"{% url 'contacts.contact_invite_participants' %}", 'data-omnibox': "c-{{object.uuid}}"}
```
Comment (teehamaral) [inline templates/contacts/contact_invite_participants.haml:152]: The invite button is not visible

Screen Shot 2020-04-14 at 18 02 36

File: templates/contacts/contact_invite_participants.haml [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+                .controls
+                  - csrf_token
+                  %select#id_opt_in_flow{float:'left', data-url-template:"{% url 'flows.flow_broadcast' '0' %}"}
+                    - for flow in flows
+                      %option{value:'{"id":"{{flow.id}}", "uuid":"{{flow.uuid}}"}'}
+                        {{ flow.name }}
+                  %button#id_save_flow_btn.btn.btn-primary
+                    %trans Save
+                  %a#id_open_in_flow_editor.btn-block{'data-href-template':"{% url 'flows.flow_editor' 'uuid' %}"}
+                    %trans Open in the flow editor
```
Comment (teehamaral) [inline templates/contacts/contact_invite_participants.haml:74]: This should have cursor as `pointer`

Comment (ihor-palii) [reply templates/contacts/contact_invite_participants.haml]: Fixed styles for this link.

File: templates/contacts/contact_invite_participants.haml (hunk 1 above)
Comment (ihor-palii) [reply templates/contacts/contact_invite_participants.haml]: Changed 'Invite' link to always display.

File: templates/contacts/contact_list.haml [hunk 3]
```diff
@@ -33,6 +33,9 @@
             - if org_perms.contacts.contact_import
               %a#id_import_contacts.btn.btn-primary.btn-block{href:"{% url 'contacts.contact_import' %}"}
                 -trans "Import Contacts"
+            - if org_perms.contacts.contact_invite_participants
+              %a#id_invite_participants.btn.btn-primary.btn-block{href:"{% url 'contacts.contact_invite_participants' %}"}
```
[the remark this answers is not in the document]
Comment (ihor-palii) [reply templates/contacts/contact_list.haml]: Removed class `btn-primary`

**Comment by ihor-palii** [comment]
Have added changes to supporting the new version of start flow on invite participants page.

File: templates/contacts/contact_invite_participants.haml [hunk 4]
```diff
@@ -439,49 +450,73 @@
...
+      inviteLinks.attr("data-available", true);
     }
 
     function processInvite(url, params) {
       let token = $('input[name=csrfmiddlewaretoken]').val();
       $.ajax({
         url: url,
         type: "POST",
+        contentType: 'application/x-www-form-urlencoded',
         data: {
```
Comment (teehamaral) [inline templates/contacts/contact_invite_participants.haml:474]: I've tested and the flow is not being started to the contact, maybe it's missing the flow parameter here.

Comment (ihor-palii) [reply templates/contacts/contact_invite_participants.haml]: I've changed logic according to 1.0 Now it sends requests to the current page through `GET` with param `contact_uuid`.

**Comment by ihor-palii** [comment]
Added logic to import Opt-In flow on org initialization.

File: templates/contacts/contact_invite_participants.haml [hunk 5]
```diff
@@ -0,0 +1,479 @@
...
+
+              .control-group
+                %label.control-label
+                  %trans Opt-in flow
+
+                %form.controls{method:"POST"}
+                  - csrf_token
+                  %select#id_opt_in_flow{float:'left', name:'optin_flow_uuid', data-url-template:"{% url 'flows.flow_broadcast' '0' %}"}
+                    - for flow in flows
+                      %option{value:'{{flow.uuid}}'}
```
Comment (teehamaral) [inline templates/contacts/contact_invite_participants.haml:69]: The only thing that I noticed is that the default Opt-In flow from the organization config field should be selected here by default, and the user should be able to click on _Open in the flow editor_ and go to the flow editor

Comment (ihor-palii) [reply templates/contacts/contact_invite_participants.haml]: Have added selected attribute and fixed open in flow editor link.

**Pull request merged**